### PR TITLE
Add SO_REUSEPORT support

### PR DESCRIPTION
**Pull request by FedericoCeratto**

### CODE REVIEW

**Comment by Araq** [comment]
@dom96 PING!

File: lib/posix/posix.nim [hunk 1]
```diff
@@ -1566,6 +1566,8 @@ var
     ## Receive timeout.
   SO_REUSEADDR* {.importc, header: "<sys/socket.h>".}: cint
     ## Reuse of local addresses is supported.
+  SO_REUSEPORT* {.importc, header: "<sys/socket.h>".}: cint
```
Comment (dom96) [inline lib/posix/posix.nim]: Does this have a chance of causing compilation issues on Linux <= 3.9?

Comment (reactormonk) [inline lib/posix/posix.nim]: See travis.

Comment (dom96) [inline lib/posix/posix.nim]: I guess we're going to need to introduce a ``enableReusePort`` define. Unless somebody has a better idea?

**Comment by FedericoCeratto** [comment]
We could put in a comment on when REUSEPORT has been added: https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=c617f398edd4db2b8567a28e899a88f8f574798d

**Comment by dom96** [comment]
A comment? I'm not sure I follow.

**Comment by strcmp** [comment]
that commit is from 2013. if you meant wait for it to be merged into official linux release, i guess holding your breath at the same time isn't a good idea.

**Comment by strcmp** [comment]
nevermind, it's been apart of linux since 3.9 and that was released on April 28th, 2013. 
http://kernelnewbies.org/Linux_3.9

**Comment by dom96** [comment]
Before I merge this I want to make sure it doesn't break compatibility with older Linux kernels, can someone test this?

**Comment by Araq** [comment]
@dom96 in order to move forward, please apply this patch manually and perhaps add a ``-d:archaicLinux`` switch to keep compat with older kernels.

**Comment by dom96** [comment]
I think instead we should add a define to *enable* it, i.e. make it opt-in.

**Comment by Araq** [comment]
Meh but then who will know about this and enable it explicitly?

**Comment by dom96** [comment]
People who want to use `SO_REUSEPORT`.

**Pull request merged**